app/server/src/routes/resource/resource.validate.js

```diff
@@ -6,19 +6,21 @@ exports.validate = (method) => {
       return [
         body("name", "name doesn't exist").exists(),
         body("body", "body doesn't exist").exists(),
-        body("topic_id", "topic_id doesn't exist").exists(),
+        body("topic_id", "topic_id doesn't exist or broken").exists().isMongoId(),
       ];
     case "delete-resource":
       return [
-        body("resource_id", "resource_id doesn't exist").exists(),
+        body("resource_id", "resource_id doesn't exist or broken").exists().isMongoId(),
       ];
     case "update-resource":
       return [
-        body("resource_id", "resource_id doesn't exist").exists(),
+        body("resource_id", "resource_id doesn't exist or broken").exists().isMongoId(),
+        body("name", "name doesn't exist").exists(),
+        body("body", "body doesn't exist").exists(),
```

Comment:
I believe `update-resource` need not require name or body; only one of them might be being updated.

Reply:
Done!